README.md
# Description:
Our final project is a modeled domain of a community centre. We made a website using php and html in order to display info stored in our database. The database model was constructed from a previous milestone, and slightly improved upon as we implemented the website to contain data tables and working sql queries that the user could request. Along with displaying this information to the user, a simple GUI was made to allow the user to easily navigate the different categories (entities), and make query requests with ease. 

Our community website will keep track of all the members, Employees, parking stalls, rooms, foods, events and other community centers. This includes an Employee’s data such as salary, hours, age such that a manager could easily figure out how much salary he owns or find out who the oldest Employee is. Members can also use their members page to figure out what their assigned parking stall number is, and then find that number on the parking stall page to figure out what type of stall it is, and how long they can park for. Users can also use the events page to look up all the different events, and staff members can add/modify/delete events accordingly. Lastly users can use our foods page to find out all the vendors we have and their locations, and food vendor owners can use our food logs function to figure out what food is most popular.

# Navigating Our Project:
Our before/after screenshots of our working functionality is in the ‘~/snapshots/’ directory, they are organized accordingly.

Our SQL scripts (using MySQL) are in the ‘~/sql_scripts/’ directory. A full script to build the entire database is ‘create_insert.sql’.

All our SQL queries are in the main project folder in a file called ‘sql_list.txt’.


changes.txt
Changes we made so far:

- Took the feedback from milestone 2, create a super class table Employee so that our eats and parks membership is more efficient
- Change table names (eg. Volunteer_WorksIn to just Volunteer)
- Remove guardian name attribute for Volunteer
- Remove location attribute for Volunteer, Contractor, Staff
- Add/Modify attributes (eg. contract_time to contract_start and contract_end in Contractor table)
- add new attribute Food, timeAte to EmployeeEats and MemberEats relation
- add new attribute role to staff role
- add NOT NULL constraint to some attributes
- Remove location from primary key of Community_Centre, getting errors when declaring foreign keys
- change the char length for some attributes
- Removed the Parks relationship from ERD as assigning a parking number is equivalent
- Add attributes name, age, address to member
- Add type attribute to employee table so that we can do Nested aggregation with group-by query easier

food.php
<html>
<head>
<link rel="stylesheet" href="css/style.css">
</head>

<!-- NAVBAR-->
<ul>
    <li style="float:left"><img src="img/logo.png" alt="logo" width="165" height="87"></li>
    <li><a href="index.php">Community Centre</a></li>
    <li><a href="events.php">Events</a></li>
    <li><a href="food.php">Food</a></li>
    <li><a href="members.php">Members</a></li>
    <li><a href="parking.php">Parking</a></li>
    <li><a href="rooms.php">Rooms</a></li>
</ul>

<!-- THIS IS WHERE ALL OUR FUNCTIONALITY WILL GO, THIS IS OUR BODY 
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    <h1>Query Food Services</h1>
</div>
-->

<!-- JOIN QUERY FOR FOODSERVICES -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
<section>
	<article>
		<p align='center'>
		<form method="GET" action="food.php" style ='text-alignt:center'>
			<label for="calculate">Display food logs:</label>
			<input type="submit" name="DisplayFoodLogs" value="Display">
			<input type="submit" name="Hide" value="Clear">
		</form>
		</p>
	</article>
</section>

<?php
    if (isset($_GET['DisplayFoodLogs'])) {
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";

        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }
		
		if (isset($_GET['DisplayFoodLogs'])) {
			$sql = "SELECT f.Food_Service_Name, e.Employee_ID, e.Food, e.TimeAte FROM Food_Services f, EatsEmployee e WHERE f.Food_Service_ID=e.Food_Service_ID";
			$result = $conn->query($sql);
			if ($result->num_rows > 0) {
            // output data of each row
            echo "<table> <tr>
                    <th> Food Service Name </th>
					<th> Employee ID </th>
                    <th> Food </th>
					<th> Time Ate </th>
                    </tr>";
            while($row = $result->fetch_assoc()) {
                echo "<tr>";
                echo "<td>". $row["Food_Service_Name"]. "</td>".
					"<td>". $row["Employee_ID"]. "</td>".
					"<td>". $row["Food"]. "</td>".
                    "<td>". $row["TimeAte"]. "</td>";
                echo "</tr>";
            }
            echo "</table>";
			} else {
            echo "0 results";
			}
		}
		
		$sql = "SELECT f.Food_Service_Name, m.Membership_ID, m.Food, m.TimeAte FROM Food_Services f, EatsMembership m WHERE f.Food_Service_ID=m.Food_Service_ID;";
        $result = $conn->query($sql);
        if ($result->num_rows > 0) {
            // output data of each row
            echo "<table> <tr>
                    <th> Food Service Name </th>
					<th> Member ID </th>
                    <th> Food </th>
					<th> Time Ate </th>
                    </tr>";
            while($row = $result->fetch_assoc()) {
                echo "<tr>";
                echo "<td>". $row["Food_Service_Name"]. "</td>".
					"<td>". $row["Membership_ID"]. "</td>".
					"<td>". $row["Food"]. "</td>".
                    "<td>". $row["TimeAte"]. "</td>";
                echo "</tr>";
            }
            echo "</table>";
			} else {
            echo "0 results";
			}
        $conn->close();
    }
?>



</div>



<!-- Food Services Table -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    <h1>Food Services</h1>
    
    <!-- THIS DISPLAYS THE TABLE ON LINK OPEN -->
    <?php
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";

        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }

        $sql = "SELECT * FROM Food_Services";
        $result = $conn->query($sql);

        if ($result->num_rows > 0) {
            // output data of each row
            echo "<table> <tr>
                    <th> Food Service Name </th>
                    <th> Location </th>
                    <th> Community Centre </th>
                    <th> Food Service ID </th>
                    </tr>";
            while($row = $result->fetch_assoc()) {
                echo "<tr>";
                echo "<td>". $row["Food_Service_Name"]. "</td>".
                    "<td>". $row["Location"]. "</td>".
                    "<td>". $row["Community_Centre"]. "</td>".
                    "<td>". $row["Food_Service_ID"]. "</td>";
                echo "</tr>";
            }
            echo "</table>";
        } else {
            echo "0 results";
        }

        $conn->close();
    ?>
</div>

<!-- Eats_Employee Table -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    <h1>Employee Eats</h1>
    
    <!-- THIS DISPLAYS THE TABLE ON LINK OPEN -->
    <?php
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";

        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }

        $sql = "SELECT * FROM EatsEmployee";
        $result = $conn->query($sql);

        if ($result->num_rows > 0) {
            // output data of each row
            echo "<table> <tr>
                    <th> Employee ID </th>
                    <th> Food Service ID </th>
                    <th> Food </th>
                    <th> Time Ate </th>
                    </tr>";
            while($row = $result->fetch_assoc()) {
                echo "<tr>";
                echo "<td>". $row["Employee_ID"]. "</td>".
                    "<td>". $row["Food_Service_ID"]. "</td>".
                    "<td>". $row["Food"]. "</td>".
                    "<td>". $row["TimeAte"]. "</td>";
                echo "</tr>";
            }
            echo "</table>";
        } else {
            echo "0 results";
        }

        $conn->close();
    ?>
</div>

<!-- Eats_Membership Table -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    <h1>Member Eats</h1>
    
    <!-- THIS DISPLAYS THE TABLE ON LINK OPEN -->
    <?php
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";

        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }

        $sql = "SELECT * FROM EatsMembership";
        $result = $conn->query($sql);

        if ($result->num_rows > 0) {
            // output data of each row
            echo "<table> <tr>
                    <th> Member ID </th>
                    <th> Food Service ID </th>
                    <th> Food </th>
                    <th> Time Ate </th>
                    </tr>";
            while($row = $result->fetch_assoc()) {
                echo "<tr>";
                echo "<td>". $row["Membership_ID"]. "</td>".
                    "<td>". $row["Food_Service_ID"]. "</td>".
                    "<td>". $row["Food"]. "</td>".
                    "<td>". $row["TimeAte"]. "</td>";
                echo "</tr>";
            }
            echo "</table>";
        } else {
            echo "0 results";
        }

        $conn->close();
    ?>
</div>
</html>



index.php
<html>
<head>
<link rel="stylesheet" href="css/style.css">
</head>

<!-- NAVBAR-->
<ul>
    <li style="float:left"><img src="img/logo.png" alt="logo" width="165" height="87"></li>
    <li><a href="index.php">Community Centre</a></li>
    <li><a href="events.php">Events</a></li>
    <li><a href="food.php">Food</a></li>
    <li><a href="members.php">Members</a></li>
    <li><a href="parking.php">Parking</a></li>
    <li><a href="rooms.php">Rooms</a></li>
</ul>

<!-- THIS IS WHERE ALL OUR FUNCTIONALITY WILL GO, THIS IS OUR BODY -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    <h1>Welcome to Vancouver Community Centre</h1>
</div>

<div align = 'center'>
    <img src="img/community-centre.jpg" alt="Vancouver Community Centre">
</div>
<br></br>
<div align = 'center'>
    <img src="img/map.jpg" alt="map" width="600" height="400">
</div>

<!-- Projection Query for Community_Centre table -->
<div style="padding-top:60px;background-color:#ffffff;">

<form action="index.php" method="get">
    <div align='center' class="container">
      <h2>Show all other Community Centers</h2>
      <p> Please Select the Information you want to be displayed </p>
      <form>
        <label class="checkbox-inline">
          <input type="checkbox" name='Name' value="Name">Name
        </label>
        <label class="checkbox-inline">
          <input type="checkbox" name='Location' value="Location">Location
        </label>
        <label class="checkbox-inline">
          <input type="checkbox" name='Open_Time' value="Open_Time">Open Time
        </label>
        <label class="checkbox-inline">
          <input type="checkbox" name='Close_Time' value="Close_Time">Close Time
        </label>
        <input type="submit" name="ProjectionQuery" value="Show Table">
        <input type="submit" name="clear" value="Clear">
      </form>
    </div>
</form>


<!-- MAIN TABLE -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    <?php
        if (isset($_GET['ProjectionQuery'])) {
            $servername = "localhost";
            $username = "root";
            $password = "root";
            $dbname = "community-centre";

            // Create connection
            $conn = new mysqli($servername, $username, $password, $dbname);
            // Check connection
            if ($conn->connect_error) {
                die("Connection failed: " . $conn->connect_error);
            }

            $attributes ='';

            if(isset($_GET['Name'])){
                $attributes .= "Name,";
            }
            if(isset($_GET['Location'])){
                $attributes .= "Location,";
            }
            if(isset($_GET['Open_Time'])){
                $attributes .= "Open_Time,";
            }
            if(isset($_GET['Close_Time'])){
                $attributes .= "Close_Time,";
            }

            $attributes = substr($attributes, 0, -1);

            if(!empty($attributes)) {
                $sql = "SELECT $attributes FROM Community_Centre";
                $result = $conn->query($sql);

                if ($result->num_rows > 0) {
                    echo "<h1>Community Centres</h1>";

                    $table_HTML = "<table> <tr>";
                    if(isset($_GET['Name'])){
                        $table_HTML .= "<th> Name </th>";
                    }
                    if(isset($_GET['Location'])){
                        $table_HTML .= "<th> Location </th>";
                    }
                    if(isset($_GET['Open_Time'])){
                        $table_HTML .= "<th> Open Time </th>";
                    }
                    if(isset($_GET['Close_Time'])){
                        $table_HTML .= "<th> Close Time </th>";
                    }
                    $table_HTML .= "</tr>";

                    echo "$table_HTML";
                            
                    while($row = $result->fetch_assoc()) {
                        echo "<tr>";
                        if(!empty($row["Name"])) {
                            echo "<td>". $row["Name"]. "</td>"; 
                        }
                        if(!empty($row["Location"])) {
                            echo "<td>". $row["Location"]. "</td>";
                        }
                        if(!empty($row["Open_Time"])) {
                            echo "<td>". $row["Open_Time"]. "</td>";
                        }
                        if(!empty($row["Close_Time"])) {
                            echo "<td>". $row["Close_Time"]. "</td>";
                        }
                        echo "</tr>";
                    }
                    echo "</table>";
                } else {
                    echo "0 results";
                }
            }

            else {
                echo "Please select some columns";
            }

            $conn->close();
        }
    ?>
    
</div>


</html>


rooms.php
<html>
<head>
<link rel="stylesheet" href="css/style.css">
<style>
.collapsible {
  background-color: rgba(98, 167, 68, 0.2);
  color: white;
  cursor: pointer;
  padding: 10px;
  width: 100%;
  border: none;
  text-align: left;
  outline: none;
  font-size: 15px;
}

.active, .collapsible:hover {
  background-color: rgba(98, 167, 68, 0.4);
}

.content {
  padding: 0 10px;
  max-height: 0;
  overflow: hidden;
  transition: max-height 0.2s ease-out;
  background-color: #f1f1f1;
}

</style>
</head>

<!-- NAVBAR-->
<ul>
    <li style="float:left"><img src="img/logo.png" alt="logo" width="165" height="87"></li>
    <li><a href="index.php">Community Centre</a></li>
    <li><a href="events.php">Events</a></li>
    <li><a href="food.php">Food</a></li>
    <li><a href="members.php">Members</a></li>
    <li><a href="parking.php">Parking</a></li>
    <li><a href="rooms.php">Rooms</a></li>
</ul>

<!-- THIS IS WHERE ALL OUR FUNCTIONALITY WILL GO, THIS IS OUR BODY -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
<section>
    <article>
		<p align='center'>
		<form method="GET" action="rooms.php" style ='text-alignt:center'>
			<label for="calculate">Display Staff That Reserved All Rooms:</label>
			<input type="submit" name="DisplayReservedAll" value="Display">
			<input type="submit" name="Hide" value="Clear">
		</form>
		</p>
	</article>
</section>


<?php
    if (isset($_GET['DisplayReservedAll'])) {
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";

        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }
		
		if (isset($_GET['DisplayReservedAll'])) {
			// $sql = "SELECT r.Type, r.Size, r.Room_ID FROM Room r
            //             WHERE NOT EXISTS (SELECT s.EmployeeID FROM Staff s WHERE NOT EXISTS (
            //                 SELECT q.ReservedBy FROM Room q
            //                     WHERE q.ReservedBy = r.ReservedBy)";
            
            $sql = "SELECT e.Name, e.Employee_ID FROM Employee e
                        WHERE NOT EXISTS (SELECT * FROM Room r WHERE NOT EXISTS (
                            SELECT q.ReservedBy FROM Room q 
                                WHERE q.ReservedBy = Employee_ID AND q.Room_ID = r.Room_ID))";

			$result = $conn->query($sql);
			if ($result->num_rows > 0) {
            // output data of each row
            echo "<table> <tr>
                    <th> Name </th>
					<th> Employee_ID </th>
                    </tr>";
            while($row = $result->fetch_assoc()) {
                echo "<tr>";
                echo "<td>". $row["Name"]. "</td>".
					"<td>". $row["Employee_ID"]. "</td>";
                echo "</tr>";
            }
            echo "</table>";
			} else {
            echo "0 results";
			}
		}
        $conn->close();
    }
?>

</div>

<?php

    $conn = Null;

    // creates a connection
    function connect() {
        global $conn;
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";
        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            return false;
            die("Connection failed: " . $conn->connect_error);
        } else {return true;}
    }

    // closes the connection
    function disconnect() {
        global $conn;
        mysqli_close($conn);
    }

    function handleUpdateRequest() {
        global $conn;

        $Column = $_POST['updtColumn'];
        $RoomID = $_POST['updtRoomID'];
        $Modifier = $_POST['updtMod'];

        $sql = "UPDATE Room SET $Column = '$Modifier' WHERE Room_ID = '$RoomID'";

        $conn->query($sql);
    }


    // big switch for post functions (insert, update, deletes)
    function handlePOSTRequest() {
        if (connect()) {
            if (array_key_exists('updateRequest', $_POST)) {
                handleUpdateRequest();
            }
        }
        disconnect();
    }

    if (isset($_POST['postRequest'])) {
        handlePOSTRequest();
    }

?>


<!-- Rooms Table -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">

    <!--Updating Events -->
    <button type="button" class="collapsible">Update Room</button>
    <div class="content">
        <form method="POST" action="rooms.php"> <!--refresh page when submitted-->
            <input type="hidden" id="updateRequest" name="updateRequest">
            <table><tr>
                <td> Column </td>
                <td> Room ID </td>
                <td> Modification </td>
            </tr>
            <tr>
                <td> 
                    <select name="updtColumn"> 
                        <option value="Type">Type</option>
                        <option value="Size">Size</option>
                        <option value="Community_Centre">Community Centre</option>
                        <option value="Is_Reserved">Is Reserved</option>
                        <option value="ReservedBy">Reserved By</option>
                    </select>
                </td>
                <td> <input type="text" name="updtRoomID"> </td>
                <td> <input type="text" name="updtMod"> </td>
            </tr></table>

            <input type="submit" value="update" name="postRequest"></p>
        </form> 
    </div>

    <h1>Rooms</h1>
    
        <!-- THIS DISPLAYS THE TABLE ON LINK OPEN -->
    <?php
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";

        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }

        $sql = "SELECT * FROM Room";
        $result = $conn->query($sql);

        if ($result->num_rows > 0) {
            // output data of each row
            echo "<table> <tr>
                    <th> Type </th>
                    <th> Size </th>
                    <th> Community Centre </th>
                    <th> Is Reserved? </th>
                    <th> Room ID </th>
                    <th> Reserved By </th>
                    </tr>";
            while($row = $result->fetch_assoc()) {
                echo "<tr>";
                echo "<td>". $row["Type"]. "</td>".
                    "<td>". $row["Size"]. "</td>".
                    "<td>". $row["Community_Centre"]. "</td>".
                    "<td>". $row["Is_Reserved"]. "</td>".
                    "<td>". $row["Room_ID"]. "</td>".
                    "<td>". $row["ReservedBy"]. "</td>";
                echo "</tr>";
            }
            echo "</table>";
        } else {
            echo "0 results";
        }

        $conn->close();
    ?>
    
</div>

<!-- Equipment Table -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    <h1>Rooms</h1>
    
        <!-- THIS DISPLAYS THE TABLE ON LINK OPEN -->
    <?php
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";

        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }

        $sql = "SELECT * FROM Equipment";
        $result = $conn->query($sql);

        if ($result->num_rows > 0) {
            // output data of each row
            echo "<table> <tr>
                    <th> Equipment ID </th>
                    <th> Community Centre </th>
                    <th> Type </th>
                    <th> In Use? </th>
                    </tr>";
            while($row = $result->fetch_assoc()) {
                echo "<tr>";
                echo "<td>". $row["Equipment_ID"]. "</td>".
                    "<td>". $row["Community_Centre"]. "</td>".
                    "<td>". $row["Type"]. "</td>".
                    "<td>". $row["In_Use"]. "</td>";
                echo "</tr>";
            }
            echo "</table>";
        } else {
            echo "0 results";
        }

        $conn->close();
    ?>
    
</div>

<!-- JavaScript for sliding page expanding thing -->
<!-- Taken/Adapted from w3schools as usual :^) -->
<script>
    var coll = document.getElementsByClassName("collapsible");
    var i;

    for (i = 0; i < coll.length; i++) {
    coll[i].addEventListener("click", function() {
        this.classList.toggle("active");
        var content = this.nextElementSibling;
        if (content.style.maxHeight){
        content.style.maxHeight = null;
        } else {
        content.style.maxHeight = content.scrollHeight + "px";
        } 
    });
    }
</script>


</html>


parking.php
<html>
<head>
<link rel="stylesheet" href="css/style.css">
</head>

<!-- NAVBAR-->
<ul>
    <li style="float:left"><img src="img/logo.png" alt="logo" width="165" height="87"></li>
    <li><a href="index.php">Community Centre</a></li>
    <li><a href="events.php">Events</a></li>
    <li><a href="food.php">Food</a></li>
    <li><a href="members.php">Members</a></li>
    <li><a href="parking.php">Parking</a></li>
    <li><a href="rooms.php">Rooms</a></li>
</ul>

<!-- THIS IS WHERE ALL OUR FUNCTIONALITY WILL GO, THIS IS OUR BODY -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    <h1>Parking Page</h1>
</div>

<div>
    <article>
        <p align='center'>
        <form method="GET" action="parking.php" style='text-align:center'>
          <label for="calculate">Show </label>
          <select name="attribute" id="attribute">
            <option value="Parking_Num">Parking Number</option>
            <option value="Duration">Duration</option>
          </select>
          <select name="operation" id="operation">
            <option value=">">></option>
            <option value="<"><</option>
            <option value="=">=</option>
            <option value=">=">>= </option>
            <option value="<="><= </option>
          </select>
          <input type="number" id="restraint" name="restraint" step=".01" required>
          <input type="submit" name="SelectionQuery" value="Submit">
        </form>
        <form method="GET" action="parking.php" style='text-align:center'>
            <input type="submit" name="Show All" value="Show All">
        </form>
        </p>
    </article>
</div>


<!-- Parking Stalls Table -->
<div style="padding:20px;margin-top:30px;background-color:#ffffff;">
    
        <!-- THIS DISPLAYS THE TABLE ON LINK OPEN -->
    <?php
        $servername = "localhost";
        $username = "root";
        $password = "root";
        $dbname = "community-centre";

        // Create connection
        $conn = new mysqli($servername, $username, $password, $dbname);
        // Check connection
        if ($conn->connect_error) {
            die("Connection failed: " . $conn->connect_error);
        }

        if (isset($_GET['SelectionQuery'])) {
            // output data of each row
            $attribute = $_GET['attribute'];
            $operation = $_GET['operation'];
            $number = $_GET['restraint'];
            $sql = "SELECT * FROM Parking_Stall WHERE $attribute $operation $number";
            $result = $conn->query($sql);
            if ($result->num_rows > 0) {
                echo "<h1>Parking Stalls</h1>";
                echo "<p>Selection Query</p>";
                echo "<table> <tr>
                            <th> Parking Number </th>
                            <th> Duration </th>
                            <th> Type </th>
                            <th> Community Centre </th>
                            </tr>";
                    while($row = $result->fetch_assoc()) {
                        echo "<tr>";
                        echo "<td>". $row["Parking_Num"]. "</td>".
                            "<td>". $row["Duration"]. "</td>".
                            "<td>". $row["Type"]. "</td>".
                            "<td>". $row["Community_Centre"]. "</td>";
                        echo "</tr>";
                    }
                    echo "</table>";
            }
            else {
                echo "No Parking Stalls Found :(";
            }
        }

        else {
            $sql = "SELECT * FROM Parking_Stall";
            $result = $conn->query($sql);
            if ($result->num_rows > 0) {
                echo "<h1>Parking Stalls</h1>";
                echo "<p>Selection Query</p>";
                // output data of each row
                echo "<table> <tr>
                        <th> Parking Number </th>
                        <th> Duration </th>
                        <th> Type </th>
                        <th> Community Centre </th>
                        </tr>";
                while($row = $result->fetch_assoc()) {
                    echo "<tr>";
                    echo "<td>". $row["Parking_Num"]. "</td>".
                        "<td>". $row["Duration"]. "</td>".
                        "<td>". $row["Type"]. "</td>".
                        "<td>". $row["Community_Centre"]. "</td>";
                    echo "</tr>";
                }
                echo "</table>";
            } else {
                echo "0 results";
            }
        }

        $conn->close();
    ?>
    
</div>

</html>
